tests/benchmark/README.md

```diff
@@ -0,0 +1,320 @@
+### Remarks
+* `indices` and `queries` vectors needs to be explicitly initialized for a specific type. For example:
+```asm
+template <>
+std::vector<std::vector<float>> BM_VecSimIndex<fp32_index_t>::queries{};
+
+template <>
+std::vector<VecSimIndex *> BM_VecSimIndex<fp32_index_t>::indices{};
+```
+### class BM_VecSimCommon<index_type_t> : public BM_VecSimIndex<index_type_t>
+Defined in bm_common.h  
+In this class we define the test routines common to the **basics** benchmark and the [**updated index**](#bm_vecsimupdatedindex) benchmarks:
+* `TopK_BF`
+* `TopK_HNSW`
+* `Memory_FLAT` 
+* `Memory_HNSW`  
+### Remarks
+These function has `index_offset` parameter (defaults to 0) to be used when we want to apply the function on an updated index.  
+In addition, in this file we define macros with the arguments for the `TopK` registration.
+
+### class BM_VecSimBasics<index_type_t> : public BM_VecSimCommon<index_type_t>
+Defined in bm_vecsim_basics.h  
+In this class we define additional tests for the basic tests set:
+* `AddLabel` - Add one label per iteration (meaning 1 vector for single value index, and multiple vector for multi value index)
+* `DeleteLabel<algo_t>` - Delete one label per iteration (meaning 1 vector for single value index, and multiple vector for multi value index).  
+`algo_t` is a specific index type (including multi/single) because we call `GetDataByLabel()`, which is only known to HNSW_Single/Multi and BruteForce_Single/Multi. This why this test case is defined separately in each .cpp file using `DEFINE_DELETE_LABEL` macro (defined in bm_vecsim_basics.h). The registration is done like all other test cases in the initialization file.  
+* `Range_BF` 
+* `Range_HNSW`  
+In addition, in this file we define macros with the arguments for the `Range`, `AddLabel` and `DeleteLabel` registration.
+### Remarks
+The [basic benchmark set](#bm_vecsimbasics) includes functions defined both in `BM_VecSimCommon` and `BM_VecSimBasics`. Pay attention to refer the correct class when defining the test in the [initialization file](#how-to-define-and-register-a-new-test).
+### class BM_VecSimUpdatedIndex<index_type_t> : public BM_VecSimCommon<index_type_t>
+Defined in bm_updated_index.h  
+In this class constructor we add the additional indices, the "updated" indices. The "before" indices are initialized in `BM_VecSimIndex`.  
+This class defines no new test routines and uses only tests defined in `BM_VecSimCommon`.  
+### Remarks
+The constructor is called after we already defined the tests resides in `BM_VecSimCommon`, so `ref_count` is not zero and we cant rely on it to decide weather we should initialize the indices or not. This is why we have to hold another the `is_initialized` flag.  
+Also, we keep the value of `ref_count` at the moment of initialization in `first_updatedBM_ref_count` to free the indices when `ref_count` decreased to this value.
```

Comment:
Not clear to me...

Reply:
moved to code (bm_updated_index.h) lmk if it's better explained now